libraries/chain/database.cpp

```diff
@@ -2283,6 +2285,23 @@ namespace golos { namespace chain {
                             unclaimed_rewards -= claim;
 
                             const auto &voter = get(itr->voter);
+
+                            if (has_hardfork(STEEMIT_HARDFORK_0_19__756)) {
+                                auto vdo_itr = vdo_idx.lower_bound(voter.name);
+                                for (; vdo_itr != vdo_idx.end() && vdo_itr->delegatee == voter.name; ++vdo_itr) {
+                                    if (vdo_itr->interest_rate == 0) {
+                                        continue;
+                                    }
+
+                                    auto delegator_claim = (claim * vdo_itr->interest_rate) / STEEMIT_100_PERCENT;
```

Comment:
You calculate `interest_rate` on full claim.
An example:
User1 delegate 10 GESTS with 80%
User2 delegate 10000 GESTS with 1%
Self vesting 1 GESTS

Total user vesting: 10011 GESTS.

An example of curator reward: 1000 GETS
User1 receive: 1000 * 80% = 800 GESTS (when he delegate only 10 GESTS - 80000%? )
User2 receive: 200 * 1% = 2 GESTS (when he delegate 10000 GESTS? )